diff --git a/fpga/generator/clkconverter.tcl b/fpga/generator/clkconverter.tcl
index e574303008010a11c44653471442273a37959b5a..198b90a78333495fd65b4403f2d72f2aa1933824 100644
--- a/fpga/generator/clkconverter.tcl
+++ b/fpga/generator/clkconverter.tcl
@@ -20,6 +20,10 @@ set_property -dict [list CONFIG.ACLK_ASYNC {1} \
       CONFIG.ID_WIDTH {4} \
       CONFIG.MI_CLK.FREQ_HZ {208333333} \
       CONFIG.SI_CLK.FREQ_HZ {10000000}] [get_ips $ipName]
+if {[string match "digilentinc.com:nexys_video:*" $boardName]} {
+    set_property -dict [list CONFIG.MI_CLK.FREQ_HZ {100000000} \
+                            CONFIG.SI_CLK.FREQ_HZ $::env(SYSTEMCLOCK)] [get_ips $ipName]
+}
 
 generate_target {instantiation_template} [get_files ./$ipName.srcs/sources_1/ip/$ipName/$ipName.xci]
 generate_target all [get_files  ./$ipName.srcs/sources_1/ip/$ipName/$ipName.xci]
