diff --git a/fpga/generator/Makefile b/fpga/generator/Makefile
index cce80f832103e42b59fe8f8a9daa6d514db13d95..f667f5e71beb1edfa960ebe78ac1e1bfa223d6ca 100644
--- a/fpga/generator/Makefile
+++ b/fpga/generator/Makefile
@@ -2,7 +2,7 @@ dst := IP
 
 all: ArtyA7
 
-.PHONY: ArtyA7 vcu118 vcu108 genesys2 nexysa7
+.PHONY: ArtyA7 vcu118 vcu108 genesys2 nexysa7 nexysvideo
 
 ArtyA7: export XILINX_PART := xc7a100tcsg324-1
 ArtyA7: export XILINX_BOARD := digilentinc.com:arty-a7-100:part0:1.1
@@ -39,13 +39,20 @@ nexysa7: export SYSTEMCLOCK := 20000000
 nexysa7: export MAXSDCCLOCK  :=  10000000
 nexysa7: FPGA_NEXYSA7
 
+nexysvideo: export XILINX_PART := xc7a200tsbg484-1
+nexysvideo: export XILINX_BOARD := digilentinc.com:nexys_video:part0:1.2
+nexysvideo: export board := nexysvideo
+nexysvideo: export SYSTEMCLOCK := 20000000
+nexysvideo: export MAXSDCCLOCK  :=  5000000
+nexysvideo: FPGA_NEXYSVIDEO
+
 # variables computed from config
 EXT_MEM_BASE = $(shell grep 'EXT_MEM_BASE' ../../config/deriv/fpga$(board)/config.vh | sed 's/.*=.*h\([[:alnum:]]*\);/0x\1/g')
 
 EXT_MEM_RANGE = $(shell grep 'EXT_MEM_RANGE' ../../config/deriv/fpga$(board)/config.vh | sed 's/.*=.*h\([[:alnum:]]*\);/\1/g' | sed 's/\(.*\)/base=16;\1+1/g' | bc | sed 's/\(.*\)/0x\1/g')
 
 
-.PHONY: FPGA_Arty FPGA_VCU
+.PHONY: FPGA_Arty FPGA_VCU FPGA_NEXYSVIDEO
 FPGA_Arty: PreProcessFiles IP_Arty zsbl
 	vivado -mode tcl -source wally.tcl 2>&1 | tee wally.log
 FPGA_VCU: PreProcessFiles IP_VCU zsbl
@@ -57,8 +64,11 @@ FPGA_GENESYS2: PreProcessFiles IP_GENESYS2 zsbl
 FPGA_NEXYSA7: PreProcessFiles IP_NEXYSA7 zsbl
 	vivado -mode tcl -source wally.tcl 2>&1 | tee wally.log
 
+FPGA_NEXYSVIDEO: PreProcessFiles IP_NEXYSVIDEO zsbl
+	vivado -mode tcl -source wally.tcl 2>&1 | tee wally.log
+
 # Generate IP Blocks
-.PHONY: IP_Arty IP_VCU
+.PHONY: IP_Arty IP_VCU IP_NEXYSVIDEO
 IP_VCU: $(dst)/sysrst.log \
 	MEM_VCU \
 	$(dst)/clkconverter.log \
@@ -81,8 +91,14 @@ IP_NEXYSA7: $(dst)/sysrst.log \
 	$(dst)/clkconverter.log \
 	$(dst)/ahbaxibridge.log
 
+IP_NEXYSVIDEO: $(dst)/sysrst.log \
+	MEM_NEXYSVIDEO \
+	$(dst)/mmcm-nexysvideo.log \
+	$(dst)/clkconverter.log \
+	$(dst)/ahbaxibridge.log
+
 # Generate Memory IP Blocks
-.PHONY: MEM_VCU MEM_Arty
+.PHONY: MEM_VCU MEM_Arty MEM_NEXYSVIDEO
 MEM_VCU:
 	$(MAKE) $(dst)/ddr4-$(board).log
 MEM_Arty:
@@ -94,6 +110,9 @@ MEM_GENESYS2:
 MEM_NEXYSA7:
 	$(MAKE) $(dst)/ddr2-$(board).log
 
+MEM_NEXYSVIDEO:
+	$(MAKE) $(dst)/ddr3-$(board).log
+
 # Copy files and make necessary modifications
 .PHONY: PreProcessFiles
 PreProcessFiles:
@@ -117,6 +136,8 @@ zsbl:
 	SYSTEMCLOCK=$(SYSTEMCLOCK) EXT_MEM_BASE=$(EXT_MEM_BASE) EXT_MEM_RANGE=$(EXT_MEM_RANGE) $(MAKE) -C ../zsbl
 
 # Generate Individual IP Blocks
+$(dst)/ddr3-nexysvideo.log: xlnx_ddr3-nexysvideo-mig.prj
+
 $(dst)/%.log: %.tcl
 	mkdir -p IP
 	cd IP;\
